### PR TITLE
6371 migrate sharding tags

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

#245

**Description**

implement delete , edit and add tag dialog

**Additional context**

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-xqyhovwfmh.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/services-ngx/entrypoint.service.spec.ts [hunk 1]
```diff
@@ -69,6 +69,23 @@ describe('EntrypointService', () => {
     });
   });
 
+  describe('update', () => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/services-ngx/entrypoint.service.spec.ts]:
```suggestion
  describe('delete', () => {
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-bhfpfbwhpa.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-sqtfdrrfix.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-evvbwtvkpu.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/services-ngx/entrypoint.service.ts [hunk 2]
```diff
@@ -33,4 +33,8 @@ export class EntrypointService {
   update(entrypoint: Entrypoint): Observable<void> {
     return this.http.put<void>(`${this.constants.org.baseURL}/configuration/entrypoints/`, entrypoint);
   }
+
+  delete(entrypointId: string): Observable<void> {
+    return this.http.delete<void>(`${this.constants.org.baseURL}/configuration/entrypoints/` + entrypointId);
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/entrypoint.service.ts]:
```suggestion
    return this.http.delete<void>(`${this.constants.org.baseURL}/configuration/entrypoints/${entrypointId}`);
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            useFactory: () => dialogData,
+          },
+          { provide: MatDialogRef, useValue: matDialogRefMock },
+        ],
+      });
+      fixture = TestBed.createComponent(OrgSettingAddTagDialogComponent);
+      component = fixture.componentInstance;
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      httpTestingController = TestBed.inject(HttpTestingController);
+    });
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.spec.ts:64]: To move outside `describe('tag creation')`?

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 4]
```diff
@@ -176,8 +178,31 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
+        role: 'dialog',
+        id: 'addTagDialog',
+      })
+      .afterClosed()
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        filter((result) => !!result),
+        switchMap((updatedTag) => this.tagService.update(updatedTag)),
+        tap(() => {
+          this.snackBarService.success('Tag successfully created!');
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
          this.snackBarService.success('Tag successfully updated!');
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.html [hunk 5]
```diff
@@ -28,6 +28,7 @@ <h2 mat-dialog-title>{{ isUpdate ? 'Edit a tag' : 'Create a tag' }}</h2>
         <mat-hint align="end">{{ tagForm.get('name').value?.length ?? 0 }}/64</mat-hint>
         <mat-error *ngIf="tagForm.get('name').hasError('maxlength')">Name has to be less than 64 characters long.</mat-error>
         <mat-error *ngIf="tagForm.get('name').hasError('minlength')">Name has to be more than 2 characters long.</mat-error>
+        <mat-error *ngIf="mappingForm.get('value').hasError('required')">Name is required.</mat-error>
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.html]:
```suggestion
        <mat-error *ngIf="tagForm.get('name').hasError('required')">Name is required.</mat-error>
```

**Review by gaetanmaisse — APPROVED** [review]

**Comment by gaetanmaisse** [comment]
Not directly related to this PR but it looks like there are some issues with tag deletion:  
 - If a tag isn't used in any entry point then no request is sent and so the tag isn't deleted
 - If an entry point has only 1 tag then when trying to delete that particular tag we got a server error: 
```json
{
  "message" : "size must be between 1 and 2147483647",
  "path" : "updateEntrypoint.arg0.tags",
  "invalid_value" : [ ]
}
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.spec.ts (hunk 3 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-add-tag-dialog.component.spec.ts:64]: i add same beforeEach for `describe('tag edition')` juste after. I did it this way for a better look in edition commit

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts (hunk 4 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]: good catch, ty,

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-jkecavrojs.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**